armbian/base/scripts/autosetup-ssd.sh

```diff
@@ -0,0 +1,205 @@
+#
+
+function usage() {
+    printf "
+BitBox Base: Auto-Setup SSD
+
+This script checks for potential storage targets and automates the setup process.
+Use with caution, data on targets will be deleted.
+
+Usage: $0 <status|format> [device] [--assume-yes]
+
+Examples:
+* $0 status                     Get list of potential target devices in system
+* $0 format /dev/sda            Set up specified device interactively
+* $0 format auto                Detect device and set up interactively
+* $0 format auto --assume-yes   WARNING! Execute only in well defined environments!
+
+"
+}
+
+function list_targets() {
+    echo
+    printf "%-16s %-7s %16s    %-30s\n" "DEVICE" "FSTYPE" "SIZE" "OK for storage?"
+    printf "%-16s %-7s %16s   %-30s\n" "---------------" "------" "-----------------" "------------------------------"
+
+    device_found=0
+    targets_total=0
+    blockdev_target_path=""
+
+    # loop over all top-level block devices
+    while read blockdev; do
+        name=$( echo "${blockdev}" | cut -f 1 -d " " )
+        type=$( echo "${blockdev}" | cut -f 2 -d " " )
+        size=$( echo "${blockdev}" | cut -f 4 -d " " )
+        blockdev_target=""
+
+        # check for existing filesystem
+        if [[ ${#type} -gt 0 ]]; then
+            blockdev_target="NO: has file system"
+        
+        # check if at least 200GB
+        elif [[ ${size} -lt 200000000000 ]]; then
+            blockdev_target="NO: too small (min 200GB)"
+
+        else
+            # check top-level device for partitions
+            partition_count=0
+            while read partition; do
+                name_part=$( echo "${partition}" | cut -f 1 -d " " )
+                if [[ "${name_part}" =~ "${name}" ]] && [[ "${name_part}" != "${name}" ]]; then
+                    partition_count=$((partition_count+1))
+                    blockdev_target="NO: has ${partition_count} partition(s)"
+                fi
+            done <<< "$(lsblk -o NAME,FSTYPE,PARTTYPE,SIZE,TYPE,MAJ:MIN -abrnp -e 1,7,31,179,252)"
+
+            if [[ ${#blockdev_target} -eq 0 ]]; then
+                targets_total=$((targets_total + 1))
+                blockdev_target="OK"
+                blockdev_target_path="${name}"
+            fi
+
+            # check if specified device is found
+            if [[ "${name}" == "${DEVICE}" ]]; then 
+                device_found=1
+            fi
+
+        fi
+
+        printf "%-16s %-7s %16s    %-20s\n" "${name}" "${type}" "${size}" "${blockdev_target}"
+
+    # `lsblk` returns only top-level devices (-d) and excludes device-types by major number
+    # (see http://www.lanana.org/docs/device-list/devices-2.6+.txt)
+    done <<< "$(lsblk -o NAME,FSTYPE,PARTTYPE,SIZE,TYPE,MAJ:MIN -abrnp -e 1,7,31,179,252 -d)"
+    
+    printf "%-16s %-7s %16s   %-20s\n" "---------------" "------" "-----------------" "------------------------------"
+    printf "TOTAL %s potential target blockdevices found\n\n" "${targets_total}"
+}
+
+
+function format() {
```

Comment:
(Optional) It might be clearer to explicitly pass arguments to the `format` function using `$1`, and using local variables inside the function:

```
function format() {
    if [ "$#" -ne 1 ]; then
        echo "usage: format <device>" >&2
        return 1
    fi
    local device="$1"
}
```

Reply:
future iteration (see below)